tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php

```diff
@@ -273,4 +275,90 @@ public function testListTableDateTypeColumns()
         $this->assertSame('datetime', $columns['col_datetime']->getType()->getName());
         $this->assertSame('datetimetz', $columns['col_datetimetz']->getType()->getName());
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createLargeNumberOfTables(): int
+    {
+        // Create a base table for test.
+        $sql = "CREATE TABLE tbl_test_2766_0 (x_id VARCHAR2(255) DEFAULT 'x' NOT NULL, x_data CLOB DEFAULT NULL NULL, x_number NUMBER(10) DEFAULT 0 NOT NULL, PRIMARY KEY(x_id))";
+        $this->_conn->executeUpdate($sql);
+
+        // Create a large number of tables with indexes and foreign keys.
+        for ($i = 1; $i < 649; $i++) {
```

Comment:
I see your point, but at the end of the day this entire issue is about performance, not about a bug.  Current code can build the required DBAL schema, if you only accept waiting looong time for it. How about increasing the accepted threshold time to, say, one  minute? Or any value that we can still consider being a failure in any possible environment? (Can anybody reasonably accept the schema is built in more than one minute, provided PHP does not throught timeouts in the meantime)